probe-rs/src/core.rs

```diff
@@ -649,6 +697,164 @@ impl<'probe> Core<'probe> {
     pub fn disable_vector_catch(&mut self, condition: VectorCatchCondition) -> Result<(), Error> {
         self.inner.disable_vector_catch(condition)
     }
+
+    /// Dumps core info with the current state.
+    ///
+    /// # Arguments
+    ///
+    /// * `stack`: The stack memory that was allocated and should be dumped.
+    /// * `heap`: The heap memory that was allocated and should be dumped.
+    /// * `additional_memory`: Additional memory ranges that should be dumped.
+    pub fn dump(&mut self, ranges: Vec<Range<u64>>) -> Result<CoreDump, Error> {
+        let instruction_set = self.instruction_set()?;
+
+        let supports_native_64bit_access = self.supports_native_64bit_access();
+
+        let mut registers = HashMap::new();
+        for register in self.registers().all_registers() {
+            let value = self.read_core_reg(register.id())?;
+            registers.insert(register.id(), value);
+        }
+
+        let mut data = Vec::new();
+        for range in ranges {
+            let mut values = vec![0; (range.end - range.start) as usize];
+            self.read_8(range.start, &mut values)?;
+            data.push((range, values));
+        }
+
+        Ok(CoreDump {
+            registers,
+            data,
+            instruction_set,
+            supports_native_64bit_access,
+        })
+    }
+}
+
+impl<'probe> CoreInterface for Core<'probe> {
```

Comment:
Why is this implementation necessary? It seems you could just get the _boxed_ `CoreInterface` out of the `Core`, and use that?